### DISCUSSION TITLE
Use of Vehicle Status as a set

### DISCUSSION THREAD

**Discussion by bevans2000** [mars-sim/mars-sim#292] [General]
The status of a Vehicle is a set of status. I have noticed that this is causing a potential double UI event when changing from PARKED to MOVING. PARKED is remove first and then MOVING is added.
It seems that the status that represent the state of the Vehicle and mutually exclusive (PARKED, GARAGED, MOVING, STUCK) and would be better represented as a primary State field. The other states such as TOWING, TOWED etc are secondary status modifiers and these should be held in a set value.

**Comment by mokun** [comment]
Yes we need to organize these flags.

Q1: what's a "double UI event" ?

> mutually exclusive (PARKED, GARAGED, MOVING, STUCK) 

Yes I agree. In that case, don't need a set.

Q2: any merit to have MAINTENANCE, UNDER_REPAIR, etc. ?

**Comment by bevans2000** [comment]
When the ui receives 2 event changes.
